src/qt/guiutil.cpp

```diff
@@ -597,20 +600,33 @@ bool SetStartOnSystemStartup(bool fAutoStart)
             TCHAR pszExePath[MAX_PATH];
             GetModuleFileName(NULL, pszExePath, sizeof(pszExePath));
 
-            TCHAR pszArgs[5] = TEXT("-min");
+            // Start client minimized
+            QString strArgs = "-min";
+            // Start testnet client, if we are currently running testnet
+            if (Params().NetworkIDString() == "test")
```

Comment:
You could generalize this by passing the current value of testnet, or even regtest to the link, 
```c++
strprintf("-testnet=%d -regtest=%d", GetBoolArg("-testnet"), GetBoolArg("-regtest"))
```
Not that I think there is much use in an autostarting regtest instance.